### PR TITLE
Restored grouping with pandas

### LINKED ISSUE #108: Bug in the grouping algorithm
The following code produces and empty DataFrame

```
import os
from pygef import Cpt

path_cpt = os.path.join(os.environ.get("DOC_PATH"), "../pygef/test_files/cpt.gef")

cpt = Cpt(path_cpt)
cpt.classify(classification="robertson", do_grouping=True, min_thickness=0.2, water_level_NAP=-10)
```

### LINKED ISSUE #130: Plot error with grouped option
Running the following code 

```python
def test_plot_classification_grouped(self):
        gef = Cpt("./tests/test_files/cpt.gef")
        gef.plot(
            show=False,
            classification="three_type_rule",
            do_grouping=True,
            min_thickness=0.2,
            water_level_NAP=-10,
        )
```

with this [cpt](https://github.com/breinbaas/pygef/blob/breinbaas/tests/test_files/cpt_2.gef)

throws an error;

```./test_plot.py::PlotTest::test_plot_classification_grouped Failed with Error: the dtypes of parameters y (object) and height (object) are incompatible
  File "c:\Users\brein\Documents\Development\Python\pygef\.env\lib\site-packages\matplotlib\axes\_axes.py", line 2381, in bar
    bottom = y - height / 2
TypeError: unsupported operand type(s) for /: 'NoneType' and 'int'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "C:\Python310\lib\unittest\case.py", line 59, in testPartExecutor
    yield
  File "C:\Python310\lib\unittest\case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "C:\Python310\lib\unittest\case.py", line 549, in _callTestMethod
    method()
  File "c:\Users\brein\Documents\Development\Python\pygef\tests\test_plot.py", line 45, in test_plot_classification_grouped
    gef.plot(
  File "c:\Users\brein\Documents\Development\Python\pygef\pygef\cpt.py", line 355, in plot
    return plot.plot_cpt(
  File "c:\Users\brein\Documents\Development\Python\pygef\pygef\plot_utils.py", line 119, in plot_cpt
    fig = add_grouped_classification(
  File "c:\Users\brein\Documents\Development\Python\pygef\pygef\plot_utils.py", line 250, in add_grouped_classification
    plt.barh(
  File "c:\Users\brein\Documents\Development\Python\pygef\.env\lib\site-packages\matplotlib\pyplot.py", line 2403, in barh
    return gca().barh(
  File "c:\Users\brein\Documents\Development\Python\pygef\.env\lib\site-packages\matplotlib\axes\_axes.py", line 2551, in barh
    patches = self.bar(x=left, height=height, width=width, bottom=y,
  File "c:\Users\brein\Documents\Development\Python\pygef\.env\lib\site-packages\matplotlib\__init__.py", line 1412, in inner
    return func(ax, *map(sanitize_sequence, args), **kwargs)
  File "c:\Users\brein\Documents\Development\Python\pygef\.env\lib\site-packages\matplotlib\axes\_axes.py", line 2383, in bar
    raise TypeError(f'the dtypes of parameters y ({y.dtype}) '
TypeError: the dtypes of parameters y (object) and height (object) are incompatible```

### PR DESCRIPTION
**Pull request by martinapippi**
The grouping functionality was not working after the update to polars. The code is restored to the pandas version since performance is not so important with this operation.

closes #130 
closes #108

### CODE REVIEW

File: pygef/grouping.py [hunk 1]
```diff
@@ -1,46 +1,45 @@
+import pandas as pd
 import polars as pl
 
 
 class GroupClassification:
     def __init__(self, zid, df, min_thickness):
+        df_group = df.to_pandas().copy()
```
Comment (tversteeg) [inline pygef/grouping.py:7]: The `.copy()` is not necessary here.

File: pygef/grouping.py [hunk 2]
```diff
@@ -1,46 +1,45 @@
...
         self.zid = zid
-        start_depth = df["depth"][0]
-        df_group = df[:, ["depth", "soil_type"]]
-
-        self.df_group = (
-            df_group.pipe(self.group_equal_layers, "soil_type", "depth", start_depth)
-            .pipe(group_significant_layers, min_thickness, start_depth)
-            .pipe(self.group_equal_layers, "layer", "zf", start_depth)
+        start_depth = df_group["depth"][0]
+        df_group = df_group.loc[:, ["depth", "soil_type"]]
```
Comment (tversteeg) [inline pygef/grouping.py:10]: Shouldn't this be the following?

```suggestion
        df_group = pl.from_pandas(df_group.loc[:, ["depth", "soil_type"]])
```

File: pygef/test_files/cpt2.gef [hunk 3]
```diff
@@ -0,0 +1,1136 @@
+#GEFID= 1, 1, 0
```
Comment (tversteeg) [inline pygef/test_files/cpt2.gef:1]: Are we allowed to include this file?

**Review by tversteeg — CHANGES_REQUESTED** [review]

File: pygef/grouping.py (hunk 2 above)
Comment (martinapippi) [reply pygef/grouping.py:10]: it's already a pandas DataFrame

File: pygef/test_files/cpt2.gef (hunk 3 above)
Comment (martinapippi) [reply pygef/test_files/cpt2.gef:1]: @breinbaas can we include the cpt file in the tests?

File: pygef/grouping.py (hunk 2 above)
Comment (tversteeg) [reply pygef/grouping.py:10]: I know, but in the next line you create a polars dataframe from it:

```python
self.df_group = pl.DataFrame(..
```

**Review by tversteeg — APPROVED** [review]

**Comment by ritchie46** [comment]
@martinapippi we try to phase out pandas. Its a bit wasteful to keep two dependencies around. What was the issue? It sounds that we need to fix the bugs and rewrite the code in another way.

**Comment by tversteeg** [comment]
> @martinapippi we try to phase out pandas. Its a bit wasteful to keep two dependencies around. What was the issue? It sounds that we need to fix the bugs and rewrite the code in another way.

I think putting a ton of effort in the grouping algorithm isn't worth the time. Especially since we want to promote the use of gef-model.

**Comment by ritchie46** [comment]
> I think putting a ton of effort in the grouping algorithm isn't worth the time. Especially since we want to promote the use of gef-model.

No need to hurry. Leave the issue open, maybe someone wants to pick it up.

**Comment by tversteeg** [comment]
> No need to hurry. Leave the issue open, maybe someone wants to pick it up.

If we apply Martina's fix, at least we don't ship broken code. And we still have the dependency on pandas anyway, since it hasn't been completely ported.

**Comment by ritchie46** [comment]
> I'm much more in favor of removing it than keeping it in while it's broken.

Me too, it does not align with the roadmap of phasing out pandas. So let's remove it.

File: pygef/test_files/cpt2.gef (hunk 3 above)
Comment (breinbaas) [reply pygef/test_files/cpt2.gef:1]: Yes public data

**Pull request merged**